arts-main/pubsub/handlers.py

```diff
@@ -81,24 +83,34 @@ def __get_runtime_or_schedule(self, msg, module):
         else:
             return self.scheduler.schedule_new_module(module)
 
-    def __create_module(self, msg):
+    def __create_module(self, msg, send_wasm=False):
         """Handle create message."""
-        data = msg.get('data')
-        if 'apis' not in data:
-            if data.get("filetype") == FileType.PY:
-                data['apis'] = ["python:python3"]
-            else:
-                data['apis'] = ["wasi:snapshot_preview1"]
-        if 'filetype' not in data:
-            if "python" in str(data.get("filename")):
-                data['filetype'] = FileType.PY
-            else:
-                data['filetype'] = FileType.WA
-
-        module = self.__object_from_dict(Module, data)
-        module.source = self.__create_or_get_file(msg)
-        module.parent = self.__get_runtime_or_schedule(msg, module)
-        module.save()
+        if(send_wasm):
```

Comment:
Since the send_wasm branch has basically nothing to do with module creation, it should probably go in `__create_module_ack`.

Reply:
It actually does. The current procedure is if the runtime responds with a "no file" result, we re-initiate create module, but with the wasm field populated this time. See https://docs.google.com/presentation/d/1aniExSAKxNyuMX7PnStiWQ3iRe0rnw2oWPNMoFN8njo/edit#slide=id.g110675c31f6_0_48 , Slide 12